### DISCUSSION TITLE
models.ts:  Generation of go structs to classes vs typescript Interfaces\types

### DISCUSSION THREAD

**Discussion by YairLevi** [wailsapp/wails#2779] [Q&A]
First, couldn't find another discussion about this for some reason. If there is one, refer me to it, please.

Is there a way to configure wails so that the structs are generated as interfaces or types in `models.ts` instead of classes?
Generally speaking, why not? Is it not more common to regard objects in js\ts as actual Objects and not classes that simply hold fields?
I thought that maybe it is a class because you could implement a function to the struct in go.
for example, the `Person` example from the docs. I tried adding a `PrintMe` method as part of the Person class, but it didn't get added in the generated `models.ts` as a function of the class `Person`, so effectively (correct me if I'm wrong), the structs in Go can be generated to a simple interface or type in `models.ts`.
Of course, I can't speak for all developers out there, but for me at least it has always been the case that I use Objects instead of fully defined classes for something that holds multiple fields.

Would like to have some answers on this, plus some feedback, if there is any, of what the library authors\maintainers think of this.

**Comment by leaanthony** [comment]
This came up in a discord chat the other day so at least one other person wants this (it was related to anonymous Go structs IIRC). I'm ok adding an option for interfaces. Perhaps typescriptify has an option already.

**Reply by YairLevi** [reply]
Sounds nice. 
Just now played with it a little, and really it seems that only the "convertValues" function is the only thing stopping me from writing:
`const obj: Type = { ... }`
and if i go around this using `Omit<Type, "convertValues">`, it acts as a simple type which is kinda fine.

I also noticed that golang's `time.Time` type is being generated as `any`. Consider me a noob asking a somewhat ignorant question - what is the problem with mapping `time.Time` type from golang to `Date` in JS/TS?

**Reply by leaanthony** [reply]
Maybe nothing. I say maybe because the process of turning a Go struct into a JS struct involves converting it to JSON and then parsing it again in JS. If you're able to do that between those 2 types then that's easy enough. For datetimes, there can be nuances between the 2 formats.

**Comment by brettinternet** [comment]
Has this been explored at all for v3 at least?

**Reply by fbbdev** [reply]
Yes it has. The new binding generator for v3 is a big step up from v2 in many ways.
Among other things, it supports a flag for turning off class generation entirely (everything is a type alias or interface).
There is also support for fine-grained control in class mode through type aliases:
```go
type ThisWillBeAClass struct {
    Field string
}

type ThisWillBeAnInterface = struct {
    Field string
}
```
